src/libcrun/container.c

```diff
@@ -806,23 +834,46 @@ container_delete_internal (libcrun_context_t *context, oci_container *def, const
 
   if (!only_cleanup && !status.detached)
     {
-      if (force)
-        {
-          ret = kill (status.pid, SIGKILL);
-          if (UNLIKELY (ret < 0) && errno != ESRCH)
-            {
-              crun_make_error (err, errno, "kill");
-              return ret;
-            }
-        }
-      else
+      if (! force)
         {
           ret = libcrun_is_container_running (&status, err);
           if (UNLIKELY (ret < 0))
             return ret;
           if (ret == 1)
             return crun_make_error (err, 0, "the container '%s' is not in 'stopped' state", id);
         }
+      else
+        {
+          cleanup_free libcrun_container_t *container = NULL;
+
+          if (def == NULL)
+            {
+              ret = read_container_config_from_state (&container, state_root, id, err);
+              if (UNLIKELY (ret < 0))
+                return ret;
+
+              def = container->container_def;
+            }
+
+          /* If the container has a pid namespace, it is enough to kill the first
+             process (pid=1 in the namespace).
+          */
+          if (! has_namespace_in_definition (def, "pid"))
```

Comment:
Can you reverse the check.
 if ( has_namespace_in_definition (def, "pid"))
...
else

Reply:
Sure.  I usually arrange the code so that the longest branch ends in the `else` to improve readability.  But in this case it doesn't change much as the difference is just two lines.